app/controllers/instructeurs/batch_operations_controller.rb

```diff
@@ -0,0 +1,37 @@
+module Instructeurs
+  class BatchOperationsController < ApplicationController
+    before_action :ensure_ownership!
+
+    def create
+      ActiveRecord::Base.transaction do
+        batch_operation = BatchOperation.create!(batch_operation_params.merge(instructeur: current_instructeur))
+        BatchOperationEnqueueAllJob.perform_later(batch_operation)
+      end
+      redirect_back(fallback_location: instructeur_procedure_url(procedure.id))
+    end
+
+    private
+
+    def batch_operation_params
+      params.require(:batch_operation)
+        .permit(:operation, dossier_ids: []).tap do |params|
+              # TODO: filter dossiers_ids out of instructeurs.dossiers.ids
+            end
+    end
+
+    def procedure_id
+      params[:procedure_id]
+    end
+
+    def procedure
+      Procedure.find(procedure_id)
+    end
+
+    def ensure_ownership!
+      if !current_instructeur.procedures.include?(procedure)
```

Comment:
```suggestion
      if !current_instructeur.procedures.exists?(@procedure.id)
```

ça permet à rails de faire une requête sql bien ciblée plutôt que charger toutes les procédures

Reply:
tx, ca sortait d'un copier/coller sur Instructeur::ProceduresController